### PR TITLE
Little refactor

### PR DESCRIPTION
**Pull request by curtisburns**
Haven't actually had a chance to test this out, but let me know your comments.

### CODE REVIEW

File: backend/functions/supermarkets/asda-delivery.js [hunk 1]
```diff
@@ -24,94 +24,103 @@ class AsdaDelivery {
...
 
-  sendEmail(startTime, endTime) {
+  async getSlots({slot_days}) {
+    // If there are no slot days, do nothing.
+    if (!slot_days) return;
+
+    // Otherwise...
+    // console.log('get slots called');
+    slot_days.forEach(day =>
+      day.slots.forEach(slot => {
```
Comment (KaneLincoln) [inline backend/functions/supermarkets/asda-delivery.js:80]: This is going to need to be `async`.

File: backend/functions/supermarkets/asda-delivery.js [hunk 2]
```diff
@@ -24,94 +24,104 @@ class AsdaDelivery {
...
-                  `${endTime.getUTCHours()}:${endTime.getUTCMinutes()}0`
-                );
-              });
-            })
-          }
-        });
-      });
+  async checkAsda(postcode) {
+    console.log('running asda', postcode);
+    const requestorigin = 'gi'; //Whats this??
```
Comment (karannavani) [inline backend/functions/supermarkets/asda-delivery.js:30]: it was part of their request headers, dunno what it does lol

File: backend/functions/supermarkets/asda-delivery.js [hunk 3]
```diff
@@ -24,94 +24,104 @@ class AsdaDelivery {
...
+    const data = {
+      service_info: {
+        fulfillment_type: 'DELIVERY',
+        enable_express: false,
+      },
+      //CB - What's this information here? Should it be hardcoded or dynamic?
+      start_date: '2020-04-25T00:00:00+01:00',
+      end_date: '2020-05-15T00:00:00+01:00',
```
Comment (karannavani) [inline backend/functions/supermarkets/asda-delivery.js:38]: nice catch, needs to change to be dynamically generated

Comment (karannavani) [reply backend/functions/supermarkets/asda-delivery.js:38]: i'll change it

**Pull request merged**